yt/convenience.py

```diff
@@ -5,119 +5,124 @@
-    try:
-        # os.PathLike is python >= 3.6
-        path_types = str, os.PathLike
-    except AttributeError:
-        path_types = str,
+def load(fn, *args, **kwargs):
+    """
+    Load a Dataset or DatasetSeries object.
+    The data format is automatically discovered, and the exact return type is the
+    corresponding subclass of :class:`yt.data_objects.static_output.Dataset`.
+    A :class:`yt.data_objects.time_series.DatasetSeries` is created if the first
+    argument is a pattern.
 
-    return [os.path.expanduser(arg) if isinstance(arg, path_types)
-            else arg for arg in args]
+    Parameters
+    ----------
+    fn : str, os.Pathlike, or byte (types supported by os.path.expandusers)
+        A path to the data location. This can be a file name, directory name, a glob
+        pattern, or a url (for data types that support it).
 
-def load(*args ,**kwargs):
-    """
-    This function attempts to determine the base data type of a filename or
-    other set of arguments by calling
-    :meth:`yt.data_objects.static_output.Dataset._is_valid` until it finds a
-    match, at which point it returns an instance of the appropriate
-    :class:`yt.data_objects.static_output.Dataset` subclass.
+    Additional arguments, if any, are passed down to the return class.
+
+    Returns
+    -------
+    Dataset
+        If fn is a single path, create a Dataset from the appropriate subclass.
+
+    DatasetSeries (Dataset container class)
+        If fn is a glob pattern (i.e. containing wildcard '[]?!*'), create a series.
+
+    Raises
+    ------
+    IOError
+        If fn does not match any existing file or directory.
+
+    YTOutputNotIdentified
+        If fn matches existing file or directory with undetermined format.
     """
-    args = _sanitize_load_args(*args)
-    candidates = []
-    valid_file = []
-    for argno, arg in enumerate(args):
-        if isinstance(arg, str):
-            if os.path.exists(arg):
-                valid_file.append(True)
-            elif arg.startswith("http"):
-                valid_file.append(True)
-            else:
-                if os.path.exists(os.path.join(ytcfg.get("yt", "test_data_dir"), arg)):
-                    valid_file.append(True)
-                    args[argno] = os.path.join(ytcfg.get("yt", "test_data_dir"), arg)
-                else:
-                    valid_file.append(False)
-        else:
-            valid_file.append(False)
-    types_to_check = output_type_registry
-    if not any(valid_file):
-        try:
-            from yt.data_objects.time_series import DatasetSeries
-            ts = DatasetSeries.from_filenames(*args, **kwargs)
-            return ts
-        except (TypeError, YTOutputNotIdentified):
-            pass
-        # We check if either the first argument is a dict or list, in which
-        # case we try identifying candidates.
-        if len(args) > 0 and isinstance(args[0], (list, dict)):
```

Comment:
This is the only functionality I have completely removed. The reason for this was that I could not find any registered output type whose key starts with `"_stream"`, so I concluded this block is dead.